tests/run-pass/no_main.stderr

```diff
@@ -0,0 +1,2 @@
+error: miri can only run programs that have a main function
```

Comment:
Hm, seems a bit odd to have this as a run-pass test.^^ I think that just happens to work as run-pass currently ignores the exit code, but IMO we should not rely on that.

Reply:
Yes that makes sense, but if I move both files to the `compile-fail` folder, the test doesn't pass

Reply:
What's the test failure that you're getting?